### ISSUE TITLE
Gram matrices and unaligned data

### ISSUE THREAD

**Issue by Pazaak** [mstrazar/mklaren#1]
Hi!
First of all I want to congratulate you for the amazing work put in this project, there are not so many stand alone MKL codes to use in Python, surprisingly.

I'm trying to use your work to apply multiple kernel regression over time-series. In case you don't know, this kind of data tends to be unaligned (not all the samples contain the same quantity of data). My form of preprocessing it is to create the Gram Matrix using several alignment kernel methods and then feed those matrices to the SVM.

I've tried several things. I'm using how alignment methods, multiplying the Gram Matrices of each kernel by the computed mu value. This, however, returns mediocre results. RidgeLowRank function only accepts Kinterfaces. I cannot create a Kinterface with the gram matrix directly, and inputing a numpy matrix with the original data of different row sizes makes the interface crash. KernelRidge suffers from the same problem. The Mklaren crashes using gram matrices. In 'cholesky_steps' function, a list called 'ina' empties before the loop ends and it crashes.

TL;DR
- RidgeLowRank only accepts Kinterfaces?
- A Kinterface can be build with numpy arrays of different row sizes?
- A Kinterface can be created with a gram matrix (with a precomputed kernel option, maybe)?
- Mklaren function seems to crash using gram matrices (tries to get argmax of a list that becomes empty)

Thank you so much for your patience.
~Pazaak

**Comment by mstrazar** [comment]
Dear Pazaak, thank you for you interest in the library! 

I see two possible solutions to this problem:

1. Provide a list of time series, along with the kernel function to create a Kinterface. This solution is currently employed e.g. when the data are strings (of potentially different lengths).

2. If solution 1. is not suitable for you, we could enable the creation of the Kinterface object using a precomputed Gram matrix. This somehow cancels the idea of computing the value of the kernel function "just-in-time" (i.e. when needed), but would provide working regression models.



Do you store you Gram matrices as normal symetric numpy arrays? How big are they? Would you be able to provide an example?

Cheers,
Martin
